p2p/5.Create_an_ERC-20/README.md

```diff
@@ -0,0 +1,245 @@
+
+### 📑 **Description**:
+
+In this step, you will put in place the necessary foundations for the creation of an ERC-20.
+
+### 📌 **Tasks**:
+
+- First, delete the `script` folder we will not need it as well as the `Counter.sol` and `Counter.t.sol` files.
+- Create a folder `interface` in the `src` folder.
+- Copy the `IERC20.sol` file which is in the `utils` folder to the `interface` folder, previously created.
+    - This file contains the ERC20 [interface](https://docs.soliditylang.org/en/v0.8.21/contracts.html#interfaces), all the functions we are going to implement.
+- Now, create the `ERC20.sol` file in `src`, it is in this file that we will create our ERC-20.
+- Finally, create the `ERC20` contract [inheriting](https://docs.soliditylang.org/en/v0.8.21/contracts.html#inheritance) the `IERC20` interface.
+
+> ⚠️ Don't forget the import and the header of the file
+
+### 📚 **Documentation**:
+
+- [Interface](https://docs.soliditylang.org/en/v0.8.21/contracts.html#interfaces)
+- [Inheritance](https://docs.soliditylang.org/en/v0.8.21/contracts.html#inheritance)
+
+## Step 2 - Let's go on
+
+### 📑 **Description**:
+
+In this step, you will create the variables, the mappings and the constructor necessary for an ERC-20.
+
+### 📌 **Tasks**:
+
+- Create in the `ERC20` contract the following variables:
+    - All variables except `_totalSupply` will be defined on declaration
+    - `name` which corresponds to the name of the token
+    - `symbol` which corresponds to the abbreviation of the token name
+    - `decimals` which corresponds to the number of decimals that the token can have
+    - `_totalSupply` which corresponds to the total token created
+> ⚠️ Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your variables! The names of the variables can give you clues 😉
+
+- Create in the `ERC20` contract the following [mappings](https://docs.soliditylang.org/en/v0.8.21/types.html#mapping-types):
+    - `_balances` wich corresponds to the balance of all wallets
+    - `_allowances` which corresponds to the number of tokens whose owner has authorized the spender to use
+> ⚠️ Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your mappings!
+
+- Initialize `_totalSupply` in the [constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor) of the contract.
+> ⚠️ Don't forget to assign `_totalSupply` to a wallet and to emit an event
+
+### 📚 **Documentation**:
+
+- [Variable visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility)
+- [Mapping](https://docs.soliditylang.org/en/v0.8.21/types.html#mapping-types)
+- [Constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor)
+
+## Step 3 - Let's get into the hard
+
+### 📑 **Description**:
+
+In this step, you will do the most important part: the implementation of the standard functions.
```

Comment:
same, add more explanations abot standard functions, even if it's juste one/two phrases.
Else the entire workshop is just instructions of what to do, but not why we do it, what‘s important about it etc...